src/v_fmt.c

```diff
@@ -837,3 +837,163 @@ int V_LumpSize(int lump)
     return lumpinfo[lump].fmt_size ? lumpinfo[lump].fmt_size
                                    : lumpinfo[lump].size;
 }
+
+static byte grayscale[] = {
+    0x00, 0x00, 0x00, 0xFF, 0xFF, 0xFF,
+    0xEF, 0xEF, 0xEF, 0xE7, 0xE7, 0xE7, 0xDF, 0xDF, 0xDF, 0xDB, 0xDB, 0xDB,
+    0xD3, 0xD3, 0xD3, 0xCB, 0xCB, 0xCB, 0xC7, 0xC7, 0xC7, 0xBF, 0xBF, 0xBF,
+    0xB7, 0xB7, 0xB7, 0xB3, 0xB3, 0xB3, 0xAB, 0xAB, 0xAB, 0xA7, 0xA7, 0xA7,
+    0x9F, 0x9F, 0x9F, 0x97, 0x97, 0x97, 0x93, 0x93, 0x93, 0x8B, 0x8B, 0x8B,
+    0x83, 0x83, 0x83, 0x7F, 0x7F, 0x7F, 0x77, 0x77, 0x77, 0x6F, 0x6F, 0x6F,
+    0x6B, 0x6B, 0x6B, 0x63, 0x63, 0x63, 0x5B, 0x5B, 0x5B, 0x57, 0x57, 0x57,
+    // 0x4F, 0x4F, 0x4F, 0x47, 0x47, 0x47, 0x43, 0x43, 0x43, 0x3B, 0x3B, 0x3B,
+    // 0x37, 0x37, 0x37, 0x2F, 0x2F, 0x2F, 0x27, 0x27, 0x27, 0x23, 0x23, 0x23
+};
+
+static byte GetGrayscaleColor(int r, int g, int b)
```

Comment:
I think I don't quite get this part. If we transform brightmap PNGs to grayscale, why do we need a lookup table for this?

Reply:
We map brighmap colors to "light levels". The `grayscale` table corresponds to `colormaps`.

Reply:
Ah! Maybe that's worth a comment here. 😉